File: android/src/main/java/com/getcapacitor/plugin/braintree/popupbridge/BraintreePopupbridgePlugin.java
package com.getcapacitor.plugin.braintree.popupbridge;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "BraintreePopupbridge")
public class BraintreePopupbridgePlugin extends Plugin {

    private BraintreePopupbridge implementation = new BraintreePopupbridge();

    public void load() {
        this.implementation.enable(
            this.getActivity(),
            this.bridge.getWebView(),
            getContext().getPackageName()
        );
    }

    @Override
    protected void handleOnResume() {
        this.implementation.handleOnResume(this.getActivity());
    }

    @Override
    protected void handleOnNewIntent(Intent intent) {
        this.implementation.handleOnNewIntent(this.getActivity(), intent);
    }

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Plugin/BraintreePopupbridgePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(BraintreePopupbridgePlugin)
public class BraintreePopupbridgePlugin: CAPPlugin {
    private let implementation = BraintreePopupbridge()
    
    override public func load() {
        self.implementation.enable(webView: self.webView!)
    }
    
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
